### PR TITLE
fix bug on bounty update

### PR DESCRIPTION
**Pull request by taitruong**
Bug in contract:
- bounty with e.g. 100 Juno
- owner updates to 50 Juno
- owner gets 50 Juno returned
- if owner accidentally sends 50 Juno on update, then 100 Juno needs to be send back - not 50

### CODE REVIEW

File: contracts/external/cw-bounties/src/contract.rs [hunk 1]
```diff
@@ -259,7 +259,9 @@ pub fn update(
         }
         Ordering::Less => {
             // If new amount is less, pay out difference to owner
-            let diff = old_amount.amount - new_amount.amount;
+            // in case owner accidentally sent funds, send back as well
+            let funds_send = may_pay(&info, &bounty.amount.denom)?;
```
Comment (taitruong) [inline contracts/external/cw-bounties/src/contract.rs:263]: This is the fix for the bug - also check test below.

File: contracts/external/cw-bounties/src/tests.rs [hunk 2]
```diff
@@ -231,195 +268,428 @@ pub fn test_close_bounty() {
...
+            balance_after.amount.u128()
+        );
+    }
 
-    // update bounty with lower amount
-    let err: ContractError = test
-        .update(
-            1,
-            coin(300, "ujuno"),
+    // case: update bounty with lower amount + owner accidentally send funds
```
Comment (taitruong) [inline contracts/external/cw-bounties/src/tests.rs:378]: Test for the fix

File: contracts/external/cw-bounties/src/tests.rs [hunk 3]
```diff
@@ -130,27 +143,42 @@ fn bounty_countract() -> Box<dyn Contract<Empty>> {
...
     test.create(
-        coin(100, "ujuno"),
+        coin(bounty_amount, JUNO_DENOM),
         "title".to_string(),
         Some("description".to_string()),
-        &[coin(100, "ujuno")],
+        &[coin(bounty_amount, JUNO_DENOM)],
     )
     .unwrap();
+    // - assert bounty
```
Comment (taitruong) [inline contracts/external/cw-bounties/src/tests.rs:160]: additional check on `Bounty` state for making sure all is fine - especially checking in tests for update and payout below.

**Comment by taitruong** [comment]
Question is whether for `Close`, `PayOut` and `UpdateOwnership` funds should also be checked and send back.

File: contracts/external/cw-bounties/src/contract.rs [hunk 4]
```diff
@@ -259,7 +259,9 @@ pub fn update(
...
+            let diff = old_amount.amount - new_amount.amount + funds_send;
```
Comment (JakeHartnell) [inline contracts/external/cw-bounties/src/contract.rs:264]:
```suggestion
            let funds_sent = may_pay(&info, &bounty.amount.denom)?;
            let diff = old_amount.amount - new_amount.amount + funds_sent;
```

**Review by JakeHartnell — APPROVED** [review]
Le nice!

**Pull request merged**